pkg/envoy/ads/response_benchmark_test.go

```diff
@@ -0,0 +1,211 @@
+	"github.com/openservicemesh/osm/pkg/signals"
+	"github.com/openservicemesh/osm/pkg/smi"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/registry"
+	configFake "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned/fake"
+	policyFake "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned/fake"
+	"github.com/openservicemesh/osm/pkg/k8s"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+var (
+	proxy           *envoy.Proxy
+	server          xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer
+	osmConfigurator *configurator.Client
+	adsServer       *Server
+)
+
+func setupTestServer(b *testing.B) {
+	stop := signals.RegisterExitHandlers()
+	msgBroker := messaging.NewBroker(stop)
+	kubeClient := k8sClientFake.NewSimpleClientset()
+	configClient := configFake.NewSimpleClientset()
+	policyClient := policyFake.NewSimpleClientset()
+	informerCollection, err := informers.NewInformerCollection(tests.MeshName, stop,
+		informers.WithKubeClient(kubeClient),
+		informers.WithConfigClient(configClient, tests.OsmMeshConfigName, tests.OsmNamespace),
+	)
+	if err != nil {
+		b.Fatalf("Failed to create informer collection: %s", err)
+	}
+	kubeController := k8s.NewKubernetesController(informerCollection, policyClient, msgBroker)
+	policyController := policy.NewPolicyController(informerCollection, kubeController, msgBroker)
+	osmConfigurator = configurator.NewConfigurator(informerCollection, tests.OsmNamespace, tests.OsmMeshConfigName, msgBroker)
+	kubeProvider := kube.NewClient(kubeController, osmConfigurator)
+
+	meshConfig := configv1alpha2.MeshConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "config.openservicemesh.io",
+			Kind:       "MeshConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: tests.OsmNamespace,
+			Name:      tests.OsmMeshConfigName,
+		}, Spec: configv1alpha2.MeshConfigSpec{
+			Certificate: configv1alpha2.CertificateSpec{
+				CertKeyBitSize:              2048,
+				ServiceCertValidityDuration: "1h",
+			},
+			Traffic: configv1alpha2.TrafficSpec{
+				EnableEgress:                      true,
+				EnablePermissiveTrafficPolicyMode: true,
+			},
+			Observability: configv1alpha2.ObservabilitySpec{
+				EnableDebugServer: false,
+				Tracing: configv1alpha2.TracingSpec{
+					Enable: false,
+				},
+			},
+			FeatureFlags: configv1alpha2.FeatureFlags{
+				EnableWASMStats:    false,
+				EnableEgressPolicy: false,
+			},
+		},
+	}
+	_, err = configClient.ConfigV1alpha2().MeshConfigs(tests.OsmNamespace).Create(context.Background(), &meshConfig, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create mesh config: %v", err)
```

Comment:
nit: might be a little more "idiomatic go" to return an error and let the test itself determine if it's a fatal error or not